components/apply/SelfIdentification.tsx

```diff
@@ -0,0 +1,197 @@
+import { FC, useState } from "react";
+import SelectInput from "@components/common/SelectInput";
+import TextInput from "@components/common/TextAreaInput";
+
```

Comment:
nit: Let's make a TS type for these const objects and use them in the state hook declarations.

Reply:
I'm not too clear on what you mean by this. Are you suggesting to abstract a component as a type and set a const var as this type? If so, what does that do with using them in the state hook declarations?

Reply:
My bad, I just meant to declare some Typescript types for the objects. For example:

```
type GenderIdentity = {
  readonly label: string;
  readonly value: string;
}

type Ethnicity = {
  readonly label: string;
  readonly value: string;
}

const GENDER_IDENTITIES: GenderIdentity[] = [
  ...
]

const ETHNICITIES: Ethnicity[] = [
  ...
]
```

and then in the React component, we can use the types for the hooks:

```
const [gender, setGender] = useState<GenderIdentity["value"]>("");
```

It's pretty much just for readability; you can also just type the state as a `string` instead of `GenderIdentity["value"]`